src/main/php/com/handlebarsjs/HandlebarsEngine.class.php

```diff
@@ -23,42 +23,41 @@
  * @see   http://handlebarsjs.com/
  */
 class HandlebarsEngine {
-  protected $mustache, $templates;
-  protected $builtin= [];
+  protected static $builtin;
+  protected $parser, $templates;
+  public $helpers;
+
+  static function __static() {
+    self::$builtin= [
+      'this'    => function($node, $context, $options) {
+        $variable= $context->lookup(null);
+        if ($context->isHash($variable) || $context->isList($variable)) {
+          return current($context->asTraversable($variable));
+        } else {
+          return $variable;
+        }
+      },
+      'lookup'  => function($node, $context, $options) {
+        return $options[0][$options[1]] ?? null;
+      },
+      '*inline' => function($node, $context, $options) {
+        $context->engine->templates->register($options[0]($node, $context, []), $node);
+      }
+    ];
+  }
```

Comment:
As these do not access `$this` they can exist once for *all* instances of `HandlebarsEngine` and need not be re-created with every instantiation.